dev/reference/performance/optimizer-hints.md

```diff
@@ -6,48 +6,252 @@ category: reference
 
 > **Note:**
 >
 > MySQL command-line clients earlier than 5.7.7 strip optimizer hints by default. If you want to use the `Hint` syntax in these earlier versions, add the `--comments` option when starting the client. For example: `mysql -h 127.0.0.1 -P 4000 -uroot --comments`.
 
-## TIDB_SMJ(t1, t2)
+## Syntax
+
+Optimizer hints are specified within `/*+ ... */` comments and follow behind the `SELECT`, `UPDATE` or `DELETE` keyword in a SQL statement. The hint names are case insensitive. If you want to use multiple hints, separate them with commas.
+
+For example, the following query uses three different hints:
+
+{{< copyable "sql" >}}
+
+```sql
+select /*+ USE_INDEX(t1, idx1), HASH_AGG(), HASH_JOIN(t1) */ count(*) from t t1, t t2 where t1.a = t2.b;
+```
+
+Currently, TiDB supports two categories of hints, which are slightly different in usage. The first category of hints is used to control the optimizer's behaviors, such as [`/*+ HASH_AGG() */`](#hash_agg); the second category of hints is used to set operation parameters for a single query, such as [`/*+ MEMORY_QUOTA(1024 MB)*/`](#memory_quotan).
+
+## Hints for controlling the optimizer
+
+A hint for controlling the optimizer follows behind **any** `SELECT`, `UPDATE` or `DELETE` keyword in a SQL statement. You can specify the applicable scope of a hint or of a table used in a hint by a **query block** introduced below. If you do not explicitly specify a query block for a hint, the hint affects the current query block by default.
+
+### Query block
+
+Each query or sub-query in a statement corresponds to a different query block, and each query block has its own `QB_NAME`. For example:
+
+{{< copyable "sql" >}}
+
+```sql
+select * from (select * from t) t1, (select * from t) t2;
+```
+
+The above query statement has three query blocks: the outermost `SELECT` corresponds to the first query block, whose `QB_NAME` is `sel_1`; the two `SELECT` sub-queries correspond to the second and the third query block, whose `QB_NAME`s are `sel_2` and `sel_3`, respectively. The sequence of the numbers is based on the appearance of `SELECT` from left to right. If you replace the first `SELECT` with `DELETE` or `UPDATE`, then the corresponding `QB_NAME` is `del_1` or `upd_1`.
+
+### QB_NAME
+
+You can specify a query block's `QB_NAME` to a value that is different from the default name. In this case, the specified `QB_NAME` and the default `QB_NAME` are both valid. For example:
+
+{{< copyable "sql" >}}
+
+```sql
+select /*+ QB_NAME(QB1) */ * from (select * from t) t1, (select * from t) t2;
+```
+
+This hint means that the `SELECT` query block's name is specified to `QB1`, which makes `QB1` and the default name `sel_1` both valid for the query block.
+
+> **Note:**
+>
+> In the above example, if the hint specifies the `QB_NAME` to `sel_2` and does not specify a new `QB_NAME` for the original second `SELECT` query block, then `sel_2` becomes an invalid name for the second `SELECT` query block.
+
+### `@QB_NAME`
```

Comment:
The inline code format isn't used for all other headings of the same level. Suggest removing it to keep the format consistent.
```suggestion
### @QB_NAME
```

Reply:
Sounds good.